pkg/scaffold/api.go

```diff
@@ -271,12 +284,33 @@ func (api *API) scaffoldV2() error {
 	return nil
 }
 
-// Since we support single group only in v2 scaffolding, validate if resource
-// being created belongs to existing group.
-func (api *API) validateResourceGroup(r *resource.Resource) error {
+// isNewGroup will return true if the group informed is diff of the pre-existents
+func (api *API) isNewGroup(r *resource.Resource) bool {
 	for _, existingGroup := range api.project.ResourceGroups() {
 		if strings.ToLower(r.Group) != strings.ToLower(existingGroup) {
-			return fmt.Errorf("group '%s' is not same as existing group '%s'. Multiple groups are not supported yet.", r.Group, existingGroup)
+			return true
+		}
+	}
+	return false
+}
+
+// validateResourceGroup will return an error if the group cannot be created
+func (api *API) validateResourceGroup(r *resource.Resource) error {
+	if api.resourceExists() {
+		return fmt.Errorf("group '%s', version '%s' and kind '%s' already exists.", r.Group, r.Version, r.Kind)
+	}
+	if api.isNewGroup(r) {
+		if api.project.Version == "1" {
+			printV1DeprecationWarning()
+			return fmt.Errorf("group '%s' is not same as existing group. Multiple groups are not supported in the v1. ", r.Group)
+		}
+		if !api.project.MultiGroup {
+			return fmt.Errorf("group '%s' is not same as existing group. Multiple groups are not enabled in this project. To enable, use the multigroup command.", r.Group)
+		}
+	}
+	for _, existingKind := range api.project.ResourceVersionKinds(r.Version) {
+		if strings.ToLower(r.Kind) == strings.ToLower(existingKind) {
+			return fmt.Errorf("kind '%s' already exists in the project for the version '%s'. Multiple groups are supported but not for the same Kinds.", r.Kind, r.Version)
```

Comment:
> Multiple groups are supported but not for the same Kinds.

Why?

Reply:
For we support it more changes are required. 
Note that the controller layout it controller/kind_controller.go. So, it currently is not supported at all.

Reply:
Implementation details should not stop us from supporting it.
We can change it. Probably change the controller file path to `controller/group_kind_controller.go` if multi-group.

Reply:
See that just adding the group will not solve the problem.
Also, we will have `KindReconciler` than it should be changed as well for GroupKindReconciler?

Reply:
Good point!

> So, IHMO it shows something which requires more details and is out of its scope. I tracked it in the past as RFE. See: #1202.

Probably it's not out-of-scope.
We have to decide what approach to take before going forward.

To be more consistent with the api directory in multi-group setup, I personally leaning towards using `controllers/<group>/<kind>_controller.go`.

@DirectXMan12 @droot Thoughts?

Reply:
Hi @mengqiy,

 I also agree that the path `controllers/<group>/<kind>_controller.go.` is more consistent and I could do the changes for we support it. Please, feel free to check.  Problem sorted out -)